components-rs/telemetry.rs

```diff
@@ -220,3 +231,127 @@ pub unsafe extern "C" fn ddog_sidecar_telemetry_add_integration_log_buffer(
     ));
     buffer.buffer.push(SidecarAction::Telemetry(action));
 }
+
+pub struct ShmCache {
+    pub last_updated: Instant,
+    pub config_sent: bool,
+    pub integrations: HashSet<String>,
+    pub composer_paths: HashSet<PathBuf>,
+    pub reader: OneWayShmReader<NamedShmHandle, CString>,
+}
+
+#[derive(Hash, Eq, PartialEq)]
+struct ShmCacheKey(String, String);
+
+impl Equivalent<ShmCacheKey> for (&String, &String) {
```

Comment:
```suggestion
impl Equivalent<ShmCacheKey> for (&str, &str) {
```
Not String - &str can be any string slice, &String means a reference to a string.